lib/Frontend/DiagnosticVerifier.cpp

```diff
@@ -786,42 +771,49 @@ void DiagnosticVerifier::autoApplyFixes(unsigned BufferID,
 /// Every time a diagnostic is generated in -verify mode, this function is
 /// called with the diagnostic.  We just buffer them up until the end of the
 /// file.
-static void VerifyModeDiagnosticHook(const llvm::SMDiagnostic &Diag,
-                                     void *Context) {
-  ((DiagnosticVerifier*)Context)->addDiagnostic(Diag);
-}
+void DiagnosticVerifier::handleDiagnostic(SourceManager &SM,
+                                          const DiagnosticInfo &Info) {
+  CapturedDiagnosticInfo capturedInfo;
+  capturedInfo.Classification = Info.Kind;
+  capturedInfo.Loc = Info.Loc;
+  if (Info.Loc.isValid()) {
+    auto lineAndColumn = SM.getLineAndColumn(Info.Loc);
+    capturedInfo.Line = lineAndColumn.first;
+    capturedInfo.Column = lineAndColumn.second;
+    capturedInfo.FileName = SM.getDisplayNameForLoc(Info.Loc);
+  }
 
+  for (auto fixIt : Info.FixIts)
+    capturedInfo.FixIts.push_back(fixIt);
```

Comment:
Nit: An unbraced loop followed by a braced block can be pretty misleading. Could you please either put braces around this loop or use `std::copy`?